infra/src/main/kotlin/fi/fta/geoviite/infra/geocoding/Geocoding.kt

```diff
@@ -65,7 +65,13 @@ data class AlignmentAddresses(
     @get:JsonIgnore
     val integerPrecisionPoints: List<AddressPoint> by lazy {
         // midPoints are even anyhow, so just transform start/end
-        listOfNotNull(startPoint.withIntegerPrecision()) + midPoints + listOfNotNull(endPoint.withIntegerPrecision())
+        val start =
+            startPoint.withIntegerPrecision()?.takeIf { s ->
+                midPoints.isEmpty() || s.address < midPoints.first().address
```

Comment:
Tämä lisächeck liittyy oikeastaan aiempaan versioon jonka päädyin revertaamaan, mutta tuntuu luontevalta checkiltä tässäkin. Periaatteessa jos alkupiste on tasametrillä niin eihän sen sitten pitäisi enää esiintyö uudelleen midpointsissa, mutta tämä siis varmistaa että tuosta ei aiheudu tuplaa.

Reply:
Onkos tässä muuten vaarana että jos `midPoints` on tyhjä että `startPoint` ja `endPoint` olisivat sama piste (tai tarpeeksi lähekkäin että ne ovat luettavissa samaksi pisteeksi?) Tällöin varmaan tarvisi palauttaa vain jompikumpi 🤔

Reply:
Tämä itse asiassa meni aika lailla uusiksi jatkokorjauksissa. Pistän niistä toisen pullarin